### ISSUE TITLE
stack trace when transport is not a currently supported transport

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#36304]
### Description of Issue/Question
When starting the minion with a transport that is not currently supported I see this stack trace:

```
[root@localhost ~]# salt-minion -ldebug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: 10.0.2.15
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] Setting up the Salt Minion "10.0.2.15"
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 232, in _bootstrap
    self.run()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 88, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 86, in minion_process
    minion.start()
  File "/usr/lib/python2.6/site-packages/salt/cli/daemons.py", line 330, in start
    super(Minion, self).start()
  File "/usr/lib/python2.6/site-packages/salt/utils/parsers.py", line 996, in start
    self.prepare()
  File "/usr/lib/python2.6/site-packages/salt/cli/daemons.py", line 315, in prepare
    import salt.daemons.flo
  File "/usr/lib/python2.6/site-packages/salt/daemons/flo/__init__.py", line 26, in <module>
    from . import core
  File "/usr/lib/python2.6/site-packages/salt/daemons/flo/core.py", line 23, in <module>
    from raet import raeting
ImportError: No module named raet
```

And my minion config looks like this:

```
[root@localhost ~]# grep -E -v '^$|#' /etc/salt/minion
master: localhost
transport: doesnotexist
```

### Steps to Reproduce Issue
1. add a transport ot your config that does not exist
2. attempt to start the minion and you will see the stack trace.

You will see this stack trace error posted above. I would expect to see an error message similar to:

```
the transport doesnotexist is currently not supported. Please use one of the following: tcp, zermoq
```

### Versions Report

```
[root@localhost ~]# salt --versions-report
Salt Version:
           Salt: 2016.9.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug 18 2016, 15:13:37)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 6.6 Final
        machine: x86_64
        release: 2.6.32-504.8.1.el6.x86_64
         system: Linux
        version: CentOS 6.6 Final
```

**Comment by rallytime** [comment]
@Ch3LL Please see #36646 for this fix.

**Comment by Ch3LL** [comment]
@rallytime i tested the fix and i'm seeing the new error message which is awesome! But i'm running into another issue at the head of 2016.3. When the transport is set incorrectly in the minion config and i try to start the minion config, it outputs the error correctly but never cleanly shuts down the master as shown below:

```
 ⚙ ch3ll@thecakeisalie  ~/git/salt   branch2016.3 <B>  sudo salt-minion -ldebug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: thecakeisalie
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] Setting up the Salt Minion "thecakeisalie"
[ERROR   ] The transport 'doesnotexist' is not supported. Please use one of the following: tcp, raet, or zeromq.
[INFO    ] The salt minion is shutting down..
[INFO    ] The Salt Minion is shut down
The salt minion is shutdown.^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C
```

As shown above I even attempted to hit Ctrl + C a bunch of times and it still did not kill the minion process. I had to put it in the background with Ctrl + z and then kill the process. 

Want me to open anew issue to track this? I believe its directly related to setting the minion transport incorrectly because that is the only time I can replicate this.

**Comment by Ch3LL** [comment]
@rallytime also did a git bisect and it looks like this commit breaks it c9c45a5d790f9e8ee259650d0b014ea5b498d4dd  

When i re-add that line from that PR it fixes it.

**Comment by rallytime** [comment]
@Ch3LL Oh, I saw that also when I was testing, but then got mixed up and thought it went away. I must have been mistaken. I am not sure if that should be a new issue or not. It is indirectly related, but I haven't made much progress on process shutdown matters lately...

@DmitryKuzmenko do you have any thoughts about that?

**Comment by Ch3LL** [comment]
I just confirmed this other issue has been resolved with pr #36745 at the head of carbon:

```
[root@9fc78783387d testing]# salt-minion -ldebug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Guessing ID. The id can be explicitly set in /etc/salt/minion
[DEBUG   ] Found minion id from generate_minion_id(): 9fc78783387d
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[INFO    ] Setting up the Salt Minion "9fc78783387d"
[ERROR   ] The transport 'doesnotexist' is not supported. Please use one of the following: tcp, raet, or zeromq.
[INFO    ] Shutting down the Salt Minion
[DEBUG   ] Stopping the multiprocessing logging queue listener
[DEBUG   ] closing multiprocessing queue
[DEBUG   ] joining multiprocessing queue thread
[DEBUG   ] Stopped the multiprocessing logging queue listener
The Salt Minion is shutdown.
[root@9fc78783387d testing]# salt --version
salt 2016.9.0-701-gb4c8573
```

Closing since everything in this issue has been resolved. Tahnks
